internal/resources/gitrepository/test_helper.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package gitrepository
```

Comment:
Why it is moved to separate file?

Reply:
The functions/variables declared within files consisting of build tags are ignored by the build tool. Since these functions/variables are used at other places also we error saying undefined. 

To overcome this issue we need to move this commonly used functions/variables to a common place. Hence, the change.